WcaOnRails/spec/models/round_spec.rb

```diff
@@ -19,4 +19,152 @@
       expect(round).to be_invalid_with_errors(format: ["'m' is not allowed for '333'"])
     end
   end
+
+  context "time limit" do
+    let(:competition) { FactoryGirl.create :competition, event_ids: %w(333 444bf 555bf) }
+    let(:round) { FactoryGirl.create :round, competition: competition, event_id: "333" }
+
+    let!(:four_blind_round) { FactoryGirl.create :round, competition: competition, event_id: "444bf", format_id: "3" }
+    let!(:five_blind_round) { FactoryGirl.create :round, competition: competition, event_id: "555bf", format_id: "3" }
+
+    it "defaults to 10 minutes" do
+      expect(round.time_limit).to eq(TimeLimit.new(centiseconds: 10*60*100, cumulative_round_ids: []))
+      expect(round.time_limit_to_s).to eq "10:00.00"
+    end
+
+    it "set to 5 minutes" do
+      round.update!(time_limit: TimeLimit.new(centiseconds: 5*60*100, cumulative_round_ids: ["333-1"]))
```

Comment:
I may be misunderstanding, but yes it does: in order to encode "3x3x3 Round 1 with a cumulative time limit of 15 minutes", you'd set the `cumulative_round_ids` for that round to `["333-1"]`.

You are right that there are definitely missing validations here... we don't validate that the rounds_ids are actually valid, nor do we check if the current round is in the list of `cumulative_round_ids` (`cumulative_round_ids` must be empty, or be an array containing the current round_id and other round_ids).